presto-redis/src/main/java/com/facebook/presto/redis/decoder/hash/ISO8601HashRedisFieldDecoder.java

```diff
@@ -51,7 +52,7 @@ public ISO8601HashRedisValueProvider(DecoderColumnHandle columnHandle, String va
         @Override
         public long getLong()
         {
-            long millis = FORMATTER.parseMillis(getSlice().toStringAscii());
+            long millis = LocalDate.from(FORMATTER.parse(getSlice().toStringAscii())).atStartOfDay().toInstant(ZoneOffset.UTC).toEpochMilli();
```

Comment:
`atStartOfDay` feels like loss of fidelity. Can you try passing in a timestamp which has hr, min in it and see if the test succeeds.

Reply:
`LocalDate` would not parse a field that has Date and time. More here - https://stackoverflow.com/questions/27454025/unable-to-obtain-localdatetime-from-temporalaccessor-when-parsing-localdatetime

We can create a custom formatter with `parseDefaulting` for the time part I guess. Please suggest

Reply:
Considering using a class like `LocalDateTime` which has time component.
https://docs.oracle.com/javase/8/docs/api/java/time/LocalDateTime.html

Reply:
Had already tried `LocalDateTime` with `DateTimeFormatter.ISO_LOCAL_DATE_TIME` as well with `DateTimeFormatter.ISO_LOCAL_DATE`

- With `ISO_LOCAL_DATE_TIME` the error is - `Text '1992-12-22' could not be parsed at index 10`
- With `ISO_LOCAL_DATE` the error is - `Unable to obtain LocalDateTime from TemporalAccessor: {},ISO,UTC resolved to 1992-11-11 of type java.time.format.Parsed`

Reply:
Using `formatter.parseBest` would fix the issue. Will update